builtin/joined_dataset.cc

```diff
@@ -181,115 +159,8 @@ struct JoinedDataset::Itl
             // We can use a fast path, since we have simple non-filtered
             // equijoin
 
-            // Where expressions for the left and right side
-            auto runSide = [&] (const AnnotatedJoinCondition::Side & side,
-                                const Dataset & dataset)
-                -> std::vector<std::tuple<ExpressionValue, RowName, RowHash> >
-                {
-                    auto generator = dataset.queryBasic
-                    (context, side.select, side.when, *side.where, side.orderBy,
-                     0, -1, true /* allowParallel */);
-                    auto rows = generator(-1);
-                
-                    if (debug)
-                        cerr << "got rows " << jsonEncode(rows) << endl;
-                    
-                    // Now we extract all values and re-sort.  This is necessary because
```

Comment:
The data model allows for multiple values for the same column with different timestamps.  It's extensively used eg in lookalike modelling, so it's not an optional extra.  In general, we can't expect that joinable rows can be put into an embedding; if they can it's an optimization but it won't always be possible.  So we probably need to have the slow and fast paths that are chosen based upon the incoming data.